usecase/src/test/java/org/mobilitydata/gtfsvalidator/usecase/ValidateShapeIdReferenceInStopTimeTest.java

```diff
@@ -0,0 +1,293 @@
+        when(mockTrip.getTripId()).thenReturn("trip id");
+        when(mockTrip.getShapeId()).thenReturn("non existing shape id");
+
+        final GtfsDataRepository mockDataRepo = mock(GtfsDataRepository.class);
+        when(mockDataRepo.getStopTimeAll()).thenReturn(mockStopTimeCollection);
+        when(mockDataRepo.getStopTimeByTripId("trip id")).thenReturn(innerMap);
+        when(mockDataRepo.getTripById("trip id")).thenReturn(null);
+        when(mockDataRepo.getTripById("trip id")).thenReturn(mockTrip);
+        when(mockDataRepo.getShapeById("non existing shape id")).thenReturn(null);
+
+        final ValidationResultRepository mockResultRepo = mock(ValidationResultRepository.class);
+        final Logger mockLogger = mock(Logger.class);
+
+        final ValidateShapeIdReferenceInStopTime underTest =
+                new ValidateShapeIdReferenceInStopTime(mockDataRepo, mockResultRepo, mockLogger);
+
+        underTest.execute();
+
+        verify(mockLogger, times(1)).info(ArgumentMatchers.eq(
+                "Validating rule 'E034 - `shape_id` must be provided and valid when" +
+                        " stop_times.shape_dist_travelled is provided'" + System.lineSeparator()));
+
+        verify(mockDataRepo, times(1)).getStopTimeAll();
+        verify(mockDataRepo, times(1))
+                .getStopTimeByTripId(ArgumentMatchers.eq("trip id"));
+        verify(mockDataRepo, times(1)).getTripById("trip id");
+        verify(mockDataRepo, times(1)).getShapeById("non existing shape id");
+
+        verify(mockStopTime, times(1)).getShapeDistTraveled();
+        verify(mockTrip, times(1)).getShapeId();
+
+        final ArgumentCaptor<NonExistingShapeNotice> captor =
+                ArgumentCaptor.forClass(NonExistingShapeNotice.class);
+
+        verify(mockResultRepo, times(1)).addNotice(captor.capture());
+
+        final List<NonExistingShapeNotice> noticeList = captor.getAllValues();
+
+        assertEquals("stop_times.txt", noticeList.get(0).getFilename());
+        assertEquals("shape_id", noticeList.get(0).getNoticeSpecific(KEY_FIELD_NAME));
+        assertEquals("trip_id", noticeList.get(0).getNoticeSpecific(KEY_COMPOSITE_KEY_FIRST_PART));
+        assertEquals("stop_sequence", noticeList.get(0).getNoticeSpecific(KEY_COMPOSITE_KEY_SECOND_PART));
+        assertEquals("trip id", noticeList.get(0).getNoticeSpecific(KEY_COMPOSITE_KEY_FIRST_VALUE));
+        assertEquals(2, noticeList.get(0).getNoticeSpecific(KEY_COMPOSITE_KEY_SECOND_VALUE));
+        assertEquals("non existing shape id", noticeList.get(0).getNoticeSpecific(KEY_UNKNOWN_SHAPE_ID));
+
+        verifyNoMoreInteractions(mockDataRepo, mockLogger, mockResultRepo, mockStopTime, mockTrip);
+    }
+
+    // suppress warning regarding ignored result of method since it is not necessary here.
+    @SuppressWarnings("ResultOfMethodCallIgnored")
+    @Test
+    void stopTimeWithNonNullShapeDistTravelledPlusValidTripIdAndNullAndValidShapeIdShouldNotGenerateNotice() {
```

Comment:
Is the shape id valid or null here? Both words are in the title

Reply:
Will rework the name of method for legibility

Reply:
To answer you question: the `shapeId` is valid here.